### PR TITLE
workaround for reverse proxy, ssh, submodule

### PR DESCRIPTION
**Pull request by likeit92**
If someone use gogs with reverse proxy and try to push a ssh submodule to the server the reference url was wrong.

This workaround works only for the local server!!!

### CODE REVIEW

File: modules/git/submodule.go [hunk 1]
```diff
@@ -51,8 +51,14 @@ func (sf *SubModuleFile) RefUrl() string {
 	i := strings.Index(url, "@")
 	j := strings.LastIndex(url, ":")
 	if i > -1 && j > -1 {
-		return "http://" + url[i+1:j] + "/" + url[j+1:]
+		// fix problem with reverse proxy works only with local server
+		if strings.Contains(appUrl,url[i+1:j]) {
```
Comment (Unknwon) [inline modules/git/submodule.go]: As a quick work around, why not just use `setting.AppUrl` here so you need to change anything in template file.

**Comment by Unknwon** [comment]
You don't need to close and reopen a pull request, GitHub detects changes automatically.

**Comment by likeit92** [comment]
okay thank you for your feedback

**Comment by Unknwon** [comment]
Travis is slow today...

**Comment by Unknwon** [comment]
OK, good, thanks!

**Pull request merged**